.github/actions/srp-helper/publish-container/publish.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Copyright 2021-2022 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+
+set -euo pipefail
+
+echo "SRP: Collect source provenance."
+
+mkdir -p ${SRP_WORKING_DIR}
+srp config auth --client-id "$CLIENT_ID" --client-secret "$CLIENT_SECRET"
+srp provenance init
+
+GITHUB_FQDN=$(echo "${GITHUB_SERVER_URL}" | sed -e "s/^https:\/\///")
+srp provenance add-build github --action ${GITHUB_ACTION} --build-id ${GITHUB_RUN_ID}_${GITHUB_RUN_ATTEMPT} --instance ${GITHUB_FQDN} --namespace ${GITHUB_REPOSITORY} --ref ${GITHUB_REF}
+srp provenance declare-source git --verbose --set-key=function-buildpack-source --path .
+srp provenance action start --name=publish
+
+srp provenance action import-cmd --cmd "make base_url=$url registry.location=other REGISTRY=$registry $target"
+observer_agent -m start_observer -e "${SRP_WORKING_DIR}"/required-envs.sh -S
+source "${SRP_WORKING_DIR}"/required-envs.sh set
+
+make base_url=$url registry.location=other REGISTRY=$registry $target
```

Comment:
The SRP docs had some mention of setting the following:
```
export GOPRIVATE="gitlab.eng.vmware.com/*"
export GOPROXY="https://build-artifactory.eng.vmware.com/artifactory/proxy-golang-remote"
export GOSUMDB="https://build-artifactory.eng.vmware.com/artifactory/go-gosumdb-remote"
```

Though I'm not sure if it applies to these build outside of VMW ecosystem

Reply:
Tested and determined that those destinations are not accessible from outside vmware